cli/src/config.rs

```diff
@@ -633,13 +645,35 @@ pub fn write_config_value_to_file(
     }
     target_table[last_key_part] = toml_edit::Item::Value(value);
 
-    // Write config back
-    std::fs::write(path, doc.to_string()).map_err(|err| {
-        user_error_with_message(
-            format!("Failed to write file {path}", path = path.display()),
-            err,
-        )
-    })
+    write_config(path, &doc)
+}
+
+pub fn remove_config_value_from_file(
+    key: &ConfigNamePathBuf,
+    path: &Path,
+) -> Result<(), CommandError> {
+    let mut doc = read_config(path)?;
+
+    // Find target table
+    let mut key_iter = key.components();
+    let last_key = key_iter.next_back().expect("key must not be empty");
+    let target_table = key_iter.try_fold(doc.as_table_mut(), |table, key| {
+        table
+            .get_mut(key)
+            .ok_or_else(|| config::ConfigError::NotFound(key.to_string()))
+            .and_then(|table| {
+                table.as_table_mut().ok_or_else(|| {
+                    config::ConfigError::Message(format!("\"{key}\" is not a table",))
```

Comment:
nit: remove trailing comma

(at your option) we can use `r#".."` to remove `\"` escapes.